ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/AdaptiveComposable.kt

```diff
@@ -27,23 +27,27 @@ internal fun RowScope.AdaptiveComposable(composables: List<@Composable () -> Uni
         modifier = Modifier
             .fillMaxWidth()
             .onGloballyPositioned { coordinates ->
-                maxSize = coordinates.size.width
+                if (coordinates.size.width != maxSize) {
+                    maxSize = coordinates.size.width
+                }
             }
             .align(Alignment.CenterVertically),
     ) {
         // Offscreen measurement
         Box(
             Modifier
-                .graphicsLayer(alpha = 0.2f)
+                .graphicsLayer(alpha = 0.0f)
                 .fillMaxWidth()
                 .wrapContentWidth(unbounded = true)
                 .align(Alignment.Center),
         ) {
             composables.forEachIndexed { index, view ->
                 Box(
                     Modifier.layout { measurable, _ ->
-                        val placeable = measurable.measure(Constraints())
-                        viewSizes[index] = placeable.width
+                        if (viewSizes[index] == 0) {
+                            viewSizes[index] = measurable.measure(Constraints()).width
+                        }
```

Comment:
As you said, I think this will cause us to not recompute the `viewSizes` when something changes (for example a locale change) since the remembered state will be kept unless the whole composable tree changes. I believe the recomposition was caused by setting a value in `viewSizes` right? If so, could we just modify this block with:
```suggestion
                        val measuredWidth =  measurable.measure(Constraints()).width
                        if (viewSizes[index] != measuredWidth) {
                            viewSizes[index] = measuredWidth
                        }
```